tharsis/resource_managed_identity_access_rules.go

```diff
@@ -0,0 +1,328 @@
+	plan.ID = types.StringValue(created.Metadata.ID)
+	plan.RunStage = types.StringValue(string(created.RunStage))
+	plan.ManagedIdentityID = types.StringValue(created.ManagedIdentityID)
+
+	allowedUsers := []types.String{}
+	for _, user := range created.AllowedUsers {
+		allowedUsers = append(allowedUsers, types.StringValue(user.Username))
+	}
+	plan.AllowedUsers = allowedUsers
+
+	allowedServiceAccounts := []types.String{}
+	for _, serviceAccount := range created.AllowedServiceAccounts {
+		allowedServiceAccounts = append(allowedServiceAccounts, types.StringValue(serviceAccount.ResourcePath))
+	}
+	plan.AllowedServiceAccounts = allowedServiceAccounts
+
+	allowedTeams := []types.String{}
+	for _, team := range created.AllowedTeams {
+		allowedTeams = append(allowedTeams, types.StringValue(team.Name))
+	}
+	plan.AllowedTeams = allowedTeams
+
+	// Set the response state to the fully-populated plan.
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (t *managedIdentityAccessRuleResource) Read(ctx context.Context,
+	req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	// Get the current state.
+	var state ManagedIdentityAccessRuleModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the managed identity access rule from Tharsis.
+	found, err := t.client.ManagedIdentity.GetManagedIdentityAccessRule(ctx,
+		&ttypes.GetManagedIdentityAccessRuleInput{
+			ID: state.ID.ValueString(),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading managed identity access rule",
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy the from-Tharsis struct to the state.
+	state.ID = types.StringValue(found.Metadata.ID)
+	state.RunStage = types.StringValue(string(found.RunStage))
+	state.ManagedIdentityID = types.StringValue(found.ManagedIdentityID)
+
+	state.AllowedUsers = []types.String{}
+	for _, user := range found.AllowedUsers {
+		state.AllowedUsers = append(state.AllowedUsers, types.StringValue(user.Username))
+	}
+
+	state.AllowedServiceAccounts = []types.String{}
+	for _, serviceAccount := range found.AllowedServiceAccounts {
+		state.AllowedServiceAccounts = append(state.AllowedServiceAccounts,
+			types.StringValue(serviceAccount.ResourcePath))
+	}
+
+	state.AllowedTeams = []types.String{}
+	for _, team := range found.AllowedTeams {
+		state.AllowedTeams = append(state.AllowedTeams, types.StringValue(team.Name))
+	}
+
+	// Set the refreshed state.
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (t *managedIdentityAccessRuleResource) Update(ctx context.Context,
+	req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	// Get the current state for its ID.
+	var state ManagedIdentityAccessRuleModel
```

Comment:
We shouldn't need to get the resource from the state here since the plan should have all the data.